scripts/generate_json_docs.py

```diff
@@ -0,0 +1,230 @@
+            source_path = inspect.getsourcefile(mod)
+            for path in sys.path:
+                source_path = source_path.replace(path, '')
+
+            source_path = source_path + "#L" + str(line)
+            method.add_source_line(source_path)
+
+            # Sketchy get examples from method docstring.
+            mod_doctest = doctest.DocTestFinder().find(mod)
+            for md in mod_doctest:
+                for example in md.examples:
+                    method.add_example(example.source)
+
+        if element.docstring:
+            if not isinstance(element, pdoc.Class) and element.cls:
+                cls = element.cls.cls
+            elif element.cls:
+                cls = element.cls
+            else:
+                cls = None
+
+            # Hack for old-style classes
+            if str(cls)[0] != '<':
+                cls = "<class '" + str(cls) + "'>"
+
+            method_info = parse_docstring(element.docstring, cls)
+
+            for name, data in method_info['arguments'].items():
+                param = Param.from_docstring_section(name, data)
+                method.add_param(param)
+
+            if method_info.get('example'):
+                method.add_example(method_info['example'])
+
+            # if 'return' in method_info:
+            if method_info.get('return'):
+                if len(method_info['return']['type_name']) > 1:
+                    method.set_returns(method_info['return']['type_name'],
+                                       method_info['return']['description'])
+
+        return method
+
+
+class Param(object):
+
+    def __init__(self, name, description=None, types=None, optional=None,
+                 nullable=None):
+        self.name = name
+        self.description = description
+        self.types = types or []
+        self.optional = optional
+        self.nullable = nullable
+
+    def to_dict(self):
+        return {'name': self.name, 'description': self.description,
+                'types': self.types, 'optional': self.optional,
+                'nullable': self.nullable}
+
+    @classmethod
+    def from_docstring_section(cls, name, data):
+        return cls(name=name, description=data['description'],
+                   types=[data['type_name']])
+
+
+def write_docs_file(path, contents):
+    if not os.path.exists(os.path.dirname(path)):
+        try:
+            os.makedirs(os.path.dirname(path))
+        except OSError:
+            raise
+    output_file = open(path, 'w')
+    output_file.write(contents)
+
+
+def generate_doc_types_json(modules, types_file_path):
+    doc_types_list = [{
+        "id": "gcloud",
+        "contents": "__init__.json",
+        "title": "__Init__"
+    }]
+
+    for module_name in modules:
+        module_path = module_name.replace('.', '/').replace('gcloud/', '')
+        doc_type_object = {
+            "id": module_path,
+            "title": " » ".join(module_path.split('/')).title(),
```

Comment:
We used to do that actually, but that logic didn't really satisfy Ruby's presentation since they display the full namespace, so we have since taken it out.